src/ripple/app/rdb/backend/RelationalDBInterfaceSqlite.cpp

```diff
@@ -720,113 +786,123 @@ RelationalDBInterfaceSqliteImp::saveValidatedLedger(
 std::optional<LedgerInfo>
 RelationalDBInterfaceSqliteImp::getLedgerInfoByIndex(LedgerIndex ledgerSeq)
 {
-    /* if database exists, use it */
     if (existsLedger())
     {
         auto db = checkoutLedger();
         return ripple::getLedgerInfoByIndex(*db, ledgerSeq, j_);
     }
 
-    /* else use shard database */
-    std::optional<LedgerInfo> res;
-    doLedger(ledgerSeq, [&](soci::session& session) {
-        res = ripple::getLedgerInfoByIndex(session, ledgerSeq, j_);
-        return true;
-    });
-    return res;
+    if (shardStoreExists())
+    {
+        std::optional<LedgerInfo> res;
+        doLedger(ledgerSeq, [&](soci::session& session) {
+            res = ripple::getLedgerInfoByIndex(session, ledgerSeq, j_);
+            return true;
+        });
+        return res;
+    }
+
+    return {};
 }
 
 std::optional<LedgerInfo>
 RelationalDBInterfaceSqliteImp::getNewestLedgerInfo()
 {
-    /* if database exists, use it */
     if (existsLedger())
     {
         auto db = checkoutLedger();
         return ripple::getNewestLedgerInfo(*db, j_);
     }
 
-    /* else use shard databases */
-    std::optional<LedgerInfo> res;
-    iterateLedgerBack(
-        {}, [&](soci::session& session, std::uint32_t shardIndex) {
-            if (auto info = ripple::getNewestLedgerInfo(session, j_))
-            {
-                res = info;
-                return false;
-            }
-            return true;
-        });
+    if (shardStoreExists())
+    {
+        std::optional<LedgerInfo> res;
+        iterateLedgerBack(
+            {}, [&](soci::session& session, std::uint32_t shardIndex) {
+                if (auto info = ripple::getNewestLedgerInfo(session, j_))
+                {
+                    res = info;
+                    return false;
+                }
+                return true;
+            });
 
-    return res;
+        return res;
+    }
+
+    return {};
 }
 
 std::optional<LedgerInfo>
 RelationalDBInterfaceSqliteImp::getLimitedOldestLedgerInfo(
     LedgerIndex ledgerFirstIndex)
 {
-    /* if database exists, use it */
     if (existsLedger())
     {
         auto db = checkoutLedger();
         return ripple::getLimitedOldestLedgerInfo(*db, ledgerFirstIndex, j_);
     }
 
-    /* else use shard databases */
-    std::optional<LedgerInfo> res;
-    iterateLedgerForward(
-        seqToShardIndex(ledgerFirstIndex),
-        [&](soci::session& session, std::uint32_t shardIndex) {
-            if (auto info = ripple::getLimitedOldestLedgerInfo(
-                    session, ledgerFirstIndex, j_))
-            {
-                res = info;
-                return false;
-            }
-            return true;
-        });
+    if (shardStoreExists())
+    {
+        std::optional<LedgerInfo> res;
+        iterateLedgerForward(
+            seqToShardIndex(ledgerFirstIndex),
+            [&](soci::session& session, std::uint32_t shardIndex) {
+                if (auto info = ripple::getLimitedOldestLedgerInfo(
```

Comment:
I don't believe it's necessary to use an `info` local. Here is a slightly optimized alternate version.
```
 if (res = ripple::getNewestLedgerInfo(session, j_); res)
     return false;
return true;
```